code/processes/reporter.q

```diff
@@ -115,10 +115,13 @@ runreport:{[tab]
 		.timer.once[startts;fp;"Reporter - ",string tab`name];
 		`..timerids upsert 1!select id,periodend:nextrun from .timer.timer where fp~/:funcparam;
 	:()];
-	/ - if the current time is within the start and end timestamps, use current time as start time, else use the startts
+        / - if timezone is specified convert the start and end timestamps to the time used by the system 
+    startts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];gtime[startts]];startts];
+    endts:?[not null tab[`timezone];.tz.ttz[`GMT;tab[`timezone];gtime[endts]];endts];
```

Comment:
Is the user has set `-localtime 0` (or not set it, which defaults to 0 I think), I think this will be wrong? `gtime` will convert the timestamp to GMT by applying the offset from local time, but if `-localtime 0`, the time is already GMT, so we don't want to do any conversion?

I think there'll need to be some sort of conditional here based on the `-localtime` setting?